doc/languages-frameworks/rust.section.md

```diff
@@ -53,10 +53,12 @@ all crate sources of this package. Currently it is obtained by inserting a
 fake checksum into the expression and building the package once. The correct
 checksum can be then take from the failed build.
 
-When the `Cargo.lock`, provided by upstream, is not in sync with the
-`Cargo.toml`, it is possible to use `cargoPatches` to update it. All patches
-added in `cargoPatches` will also be prepended to the patches in `patches` at
-build-time.
+Per the instructions in the [Cargo Book](https://doc.rust-lang.org/cargo/guide/cargo-toml-vs-cargo-lock.html)
+best practices guide, Rust applications should always commit the `Cargo.lock`
+file in git to ensure a reproducible build. However, a few packages do not, and
+Nix depends on this file, so if it missing you can use `cargoPatches` to apply
+it in the `patchPhase`. Consider sending a PR upstream with a note to the
+maintainer describing why it's important to include in the application.
```

Comment:
Aside from helping us out by letting us delete custom patches, it's also in the best interests of users/maintainers to provide a reproducible binary build and explicitly called out in the Rust/Cargo guides, so I'd imagine most maintainers would be receptive to a PR on this one.  Always worth a shot at least, particularly since `cargoPatches` already requires us to express it in patch form!